### ISSUE TITLE
Datepicker in modal breaks layout

### ISSUE THREAD

**Issue by tfr** [eternicode/bootstrap-datepicker#1496]
Hi,

I have a user interface which utilizes Bootstrap modal "popups". It's built so that in case the content of the popup is taller than the user's screen, it will extend the modal to the height of the screen and then make the content scrollable.

This works fine, except that if the content contains a datepicker, then, upon opening datepicker, it breaks badly - it breaks the scrolling and extends the modal content to its original length (through top and bottom of the screen, which, though, remains unscrollable, and also outside the modal window horizontally.

Sample:

Before clicking the datepicker (on "Purchase date"):
https://www.dropbox.com/s/vju3ppooqvushch/Screenshot%202015-07-21%2012.48.10.png?dl=0

After:
https://www.dropbox.com/s/6vujribqcl3en6k/Screenshot%202015-07-21%2012.48.15.png?dl=0

Any idea that might be causing this behaviour?

Bootstrap v3.3.0
jQuery v1.11.1
Bootstrap-datepicker latest and greatest as of today (1.5.0-dev according to package.json)

TIA,
Indrek

**Comment by hebbet** [comment]
the current dist folder in master is not up2date.
please try running grunt or use 1.4

**Comment by tfr** [comment]
I was on 1.4 before. Tried upgrading to 1.5.0-dev to see if it has been fixed.

**Comment by tfr** [comment]
Here's a test case that reproduces this bug: http://datepickerbug.codelab.solutions

**Comment by tfr** [comment]
Ah, one more thing - if you open the modal and close it immediately (without actually opening the datepicker), the datepicker will be shown after the modal has closed.

**Comment by tfr** [comment]
Ah, one more thing - if you open the modal and close it using the escape key (not the "Cancel" button at the bottom), the datepicker will be shown after the modal has closed.

**Comment by hebbet** [comment]
seems to be fixed with f08ad04, closing.

**Comment by hebbet** [comment]
@tfr can you test the current master?

**Comment by tfr** [comment]
Updated http://datepickerbug.codelab.solutions <http://datepickerbug.codelab.solutions/> - it seems slightly better in some cases, but definitely not fixed.

-- 
Indrek

**Comment by vsn4ik** [comment]
@ttr your link contains not containsactual bootstrap-datepicker (dist). Please update and re test.

**Comment by tfr** [comment]
What do you mean exactly? It contains the latest master (v1.5.0-dev) that habbet wanted me to test.

**Comment by vsn4ik** [comment]
@tfr see 7500 line in bundled js file. `Math` already removed.

**Comment by acrobat** [comment]
Yes the dist folder on master could be out of date. What you can do is, regenerate the dist folder locally and then have all the js bundled into 1 file or use the raw `bootstrap-datepicker.js` with the latest code!

The dist folder is a dev version (master = 1.5.0-dev) are only generated once in a while

**Comment by tfr** [comment]
Sorry for the delay. Finally got around to doing that (using the js/bootstrap-datepicker.js file directly from fresh clone) - unfortunately, still no cookie.

**Comment by vsn4ik** [comment]
Hmmm. If width window <= 840px, datepicker worked correctly.

**Comment by vsn4ik** [comment]
Please update Bootstrap to 3.3.5 and jQuery to 1.11.3. Maybe this is the case.

**Comment by tfr** [comment]
Vasily - good observation.

Looks like the "breaking point" is 767/768px. The framework logic is that >=768px is considered a desktop and in that case the modal desktop is centered vertically. The function that does the "magic" is BASE.centerModal(), which dynamically calculates the dialog margins and sets the margin-top parameter for the .modal-dialog class.

On mobile devices (window width <768px) the modal is on the top of the screen like Bootstrap default.

I will try to upgrade Bootstrap and jQuery, but looks like this forced margin on the dialog is the issue here.

Thanks,
Indrek

**Comment by tfr** [comment]
I patched the BASE.centerModal() function so that in case of a scrolling dialog (which is top-to-bottom on the screen) it will not set the margin anymore. Now the datepicker is displayed correctly, however, opening the datepicker undoes the stuff done in BASE.resizeContent() function, which sets the dialog height to maximum screen height and makes the .modal-body-content scrollable inside.
